vispy/visuals/text/text.py

```diff
@@ -438,7 +438,10 @@ def pos(self):
     @pos.setter
     def pos(self, pos):
         pos = [float(p) for p in pos]
-        assert len(pos) == 2
+        l = len(pos);
+        assert(l == 3 or l == 2)
+        if(l == 2):
+            pos.append(float(0.0));
```

Comment:
This assumes `pos` is a list, which you don't know. It could be tuple, it could be an ndarray, etc. Here is how I would structure it:
```
pos = np.array(pos, float)
if pos.ndim != 1 or pos.size not in (2, 3):
    raise ValueError('pos must be array-like with 2 or 3 elements')
if pos.size == 2:
    pos = np.concatenate((pos, [0.]))
```